share/search/index_strategy/trovesearch_denorm.py

```diff
@@ -0,0 +1,985 @@
+                _remaining_indexcard_pks.discard(_indexcard_pk)
+        # delete any that were skipped for any reason
+        for _indexcard_pk in _remaining_indexcard_pks:
+            yield _indexcard_pk, self.build_delete_action(_indexcard_pk)
+
+    ###
+    # implement abstract IndexStrategy.SpecificIndex
+
+    class SpecificIndex(Elastic8IndexStrategy.SpecificIndex):
+
+        # abstract method from IndexStrategy.SpecificIndex
+        def pls_handle_search__sharev2_backcompat(self, request_body=None, request_queryparams=None) -> dict:
+            return self.index_strategy.es8_client.search(
+                index=self.indexname,
+                body={
+                    **(request_body or {}),
+                    'track_total_hits': True,
+                },
+                params=(request_queryparams or {}),
+            )
+
+        # abstract method from IndexStrategy.SpecificIndex
+        def pls_handle_cardsearch(self, cardsearch_params: CardsearchParams) -> CardsearchResponse:
+            _querybuilder = _CardsearchQueryBuilder(cardsearch_params)
+            _search_kwargs = _querybuilder.build()
+            if settings.DEBUG:
+                logger.info(json.dumps(_search_kwargs, indent=2))
+            try:
+                _es8_response = self.index_strategy.es8_client.search(
+                    index=self.indexname,
+                    source=False,  # no need to get _source, identifiers are enough
+                    docvalue_fields=['card.card_iri'],
+                    highlight={
+                        'require_field_match': False,
+                        'fields': {'card.text_by_propertypath.*': {}},
+                    },
+                    **_search_kwargs,
+                )
+            except elasticsearch8.TransportError as error:
+                raise exceptions.IndexStrategyError() from error  # TODO: error messaging
+            return self.index_strategy._cardsearch_response(cardsearch_params, _es8_response, _querybuilder.cursor)
+
+        # abstract method from IndexStrategy.SpecificIndex
+        def pls_handle_valuesearch(self, valuesearch_params: ValuesearchParams) -> ValuesearchResponse:
+            _path = valuesearch_params.valuesearch_propertypath
+            _cursor = _SimpleCursor.from_page_param(valuesearch_params.page)
+            _query = (
+                _build_date_valuesearch(valuesearch_params, _cursor)
+                if is_date_property(_path[-1])
+                else _build_iri_valuesearch(valuesearch_params, _cursor)
+            )
+            if settings.DEBUG:
+                logger.info(json.dumps(_query, indent=2))
+            try:
+                _es8_response = self.index_strategy.es8_client.search(
+                    **_query,
+                    index=self.indexname,
+                )
+            except elasticsearch8.TransportError as error:
+                raise exceptions.IndexStrategyError() from error  # TODO: error messaging
```

Comment:
Minor: how annoying would it be to add error messaging instead of the `TODOs` here and above?